code/parachain/frame/assets-registry/src/tests.rs

```diff
@@ -95,37 +107,43 @@ fn update_asset() {
 		let location = <Runtime as crate::Config>::ForeignAssetId::decode(
 			&mut &XcmAssetLocation::RELAY_NATIVE.encode()[..],
 		)
-		.unwrap();
-		let ratio = rational!(42 / 123);
-		let decimals = 3;
+		.expect("Location bytes translate to foreign ID bytes");
+		let asset_info = AssetInfo {
+			name: None,
+			symbol: None,
+			decimals: 4,
+			existential_deposit: 0,
+			ratio: Some(rational!(42 / 123)),
+		};
 
 		assert_ok!(AssetsRegistry::register_asset(
 			RuntimeOrigin::root(),
-			location.clone(),
-			ratio,
-			Some(decimals)
+			LocalOrForeignAssetId::Foreign(location.clone()),
+			asset_info,
 		));
 
-		let local_asset_id = AssetsRegistry::from_foreign_asset(location.clone()).unwrap();
-		assert_eq!(
-			AssetsRegistry::from_local_asset(local_asset_id),
-			Some(ForeignMetadata { decimals: Some(decimals), location: location.clone() })
-		);
-		assert_eq!(AssetsRegistry::asset_ratio(local_asset_id), Some(ratio));
+		let local_asset_id =
+			AssetsRegistry::from_foreign_asset(location.clone()).expect("Asset exists");
+		assert_eq!(AssetsRegistry::from_local_asset(local_asset_id), Some(location.clone()));
```

Comment:
what location should set anyboy for this? Do you check that location is always interio not to break externals tokens?